### PR TITLE
fix/kuma.config can hmr

### PR DESCRIPTION
**Pull request by nojiritakeshi**
This PR is intended to reflect the contents of kuma.config.ts on the screen with hmr.

### CODE REVIEW

File: packages/system/src/generator.ts
Comment (kotarella1110) [inline packages/system/src/generator.ts:38]: Themes like `fonts` and `spacings` are increasing, and we plan to add various themes in the future as well.
As there are cases where `converter` differ for each theme token, it's not practical to process theme values all together here.
Could you please revert the changes to this code? :pray:

File: packages/system/src/color.ts
Comment (kotarella1110) [inline packages/system/src/color.ts:74]: For the same reasons as the previous comment, I would like you to revert the changes to this code as well.

**Comment by kotarella1110** [comment]
I have tested it locally, but it seems that changes to the theme are not being applied. Please check and confirm.

https://github.com/kuma-ui/kuma-ui/assets/12913947/6fcfa801-4562-4bd5-ad4d-2a7cb407083a

**Comment by nojiritakeshi** [comment]
@kotarella1110 
> I have tested it locally, but it seems that changes to the theme are not being applied. Please check and confirm.

To reproduce it locally, I think you need to specify it in the relative path of "transform,compile, StyleGenerator".

File: packages/system/src/generator.ts
Comment (kotarella1110) [reply packages/system/src/generator.ts:38]: Sorry, please wait. I checked your Discord message. I will review again after confirming.

Comment (kotarella1110) [reply packages/system/src/generator.ts:38]: This code doesn't seem to need to be reverted, so I resolve conversation.

File: packages/system/src/color.ts
Comment (kotarella1110) [inline packages/system/src/color.ts:74]: I think that [font](https://github.com/kuma-ui/kuma-ui/blob/main/packages/system/src/font.ts) etc. can also be changed.

**Comment by kotarella1110** [comment]
> To reproduce it locally, I think you need to specify it in the relative path of "transform,compile, StyleGenerator".

Could you provide more detailed explanation about this a bit further?
If possible, it would be appreciated if you could also provide the code that runs on CodeSandbox.

**Comment by nojiritakeshi** [comment]
@kotarella1110 
> Could you provide more detailed explanation about this a bit further?
If possible, it would be appreciated if you could also provide the code that runs on CodeSandbox.

Can you check it out on Github?
https://github.com/nojiritakeshi/kuma-ui/commit/154de4ffae309d8adc22693411502368b7c6d0f8

File: packages/system/src/color.ts
Comment (nojiritakeshi) [reply packages/system/src/color.ts:74]: I see that other things like fontSize have also been added!
I'll check it out and make some additional corrections.

**Comment by kotarella1110** [comment]
@nojiritakeshi 

> Can you check it out on Github?

Thanks for providing the code to reproduce it.
I tried changing to a relative path, but the theme changes were not reflected in the browser.
Even the CodeSandbox CI example doesn't seem to work. You can check the CodeSandbox CI example at the following link.

スクリーンショット 2023-08-25 21 48 40

**Comment by nojiritakeshi** [comment]
@kotarella1110 
hmm...
Can I confirm on discord?

**Comment by kotarella1110** [comment]
@nojiritakeshi Of course 👍

File: packages/vite/src/index.ts
Comment (poteboy) [inline packages/vite/src/index.ts:92]: 👍

**Comment by nojiritakeshi** [comment]
@kotarella1110 @poteboy 
Committed with modifications.
Please check back when you are available.

File: packages/compiler/src/extractor/extract.ts
Comment (kotarella1110) [inline packages/compiler/src/extractor/extract.ts:53]: @nojiritakeshi 
With this change, the HMR now works 🎉
However, I don't think it will work with Dynamic Props because the process has been moved from the system to the compiler.

```tsx
<Box p={12} color={() => "colors.blue.light"}>dynamic</Box>
```

I think it would be better to move this process to webpack-plugin and vite-plugin.

Comment (kotarella1110) [reply packages/compiler/src/extractor/extract.ts:53]: I think this process needs to be implemented on the runtime side as well.
See the following line:

https://github.com/kuma-ui/kuma-ui/blob/7ebe9ed1120b9cd5cf451c1de124fab0abd49c6a/packages/core/src/components/Box/react/utils.ts#L30

**Review by kotarella1110 — APPROVED** [review]
@nojiritakeshi Great work 🏆 Thanks!

File: packages/system/src/column.ts
Comment (kotarella1110) [inline packages/system/src/column.ts:39]: Is it possible to move the `ValueConverter` type to `system/src/types.ts` instead of removing it? I thought we should use that type here.

File: packages/system/src/border.test.ts
Comment (kotarella1110) [inline packages/system/src/border.test.ts:12]: Is it possible to keep the theme test instead of removing it?

File: packages/system/src/space.test.ts
Comment (kotarella1110) [inline packages/system/src/space.test.ts:12]: Same as above comment.

**Review by kotarella1110 — COMMENTED** [review]
@nojiritakeshi I am sorry that I approved your code, but I commented on it and you need to fix it. Once the corrections are complete, I will approve and merge again.

**Review by yukukotani — COMMENTED** [review]
Can't wait to get this landed!

File: packages/system/src/generator.ts
Comment (poteboy) [reply packages/system/src/generator.ts:36]: The reason for needing `trim()` on propValue is actually because `getFullText` is used instead of `getText` within the `collector` (My fault 🤦). I believe that fixing those area might eliminate the need for this adjustment. 
Whether or not this will be addressed in this PR is up to @nojiritakeshi  though

File: packages/system/src/border.test.ts
Comment (nojiritakeshi) [reply packages/system/src/border.test.ts:12]: @kotarella1110 
These processes would actually take place after the user theme has already been converted.
Therefore, I believe that testing the theme is unnecessary for this test

Comment (kotarella1110) [reply packages/system/src/border.test.ts:12]: @nojiritakeshi You are right. I think this line can be deleted.
However, you should keep the following test case!

https://github.com/kuma-ui/kuma-ui/pull/292/files#diff-a474cb7251af443b69bc46e96a3ce239067c318dd9f38a86cca45ef2b8020c6cL78

Comment (nojiritakeshi) [reply packages/system/src/border.test.ts:12]: @kotarella1110 
Is it correct that the code that should be left behind is the following?
`[{ borderRadius: "sm" }, "border-radius: 1rem;", ""],`

I think that the value passed as "sm" cannot be 1rem at the time this process is performed.

File: packages/system/src/generator.ts
Comment (nojiritakeshi) [reply packages/system/src/generator.ts:36]: @yukukotani @poteboy 
I'll try to fix it and include it in this PR, but I'll separate the commits.

File: packages/system/src/border.test.ts
Comment (kotarella1110) [reply packages/system/src/border.test.ts:12]: @nojiritakeshi It means that I want to keep the test case of borderRadius itself as follows.

```
[{ borderRadius: "1rem" }, "border-radius: 1rem;", ""],
```

File: packages/system/src/generator.ts
Comment (nojiritakeshi) [reply packages/system/src/generator.ts:36]: @poteboy 
I removed all the trim()'s that I thought were unnecessary, but all the existing tests were fine.
I also checked the actual operation of the system and it was fine as far as I was concerned!

Comment (poteboy) [reply packages/system/src/generator.ts:36]: Thank you 🍑

**Review by kotarella1110 — APPROVED** [review]
Thank you for your hard work 🎉🎉🎉

**Review by poteboy — APPROVED** [review]

**Pull request merged**
